x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/TransportSubmitAsyncSearchAction.java

```diff
@@ -187,7 +188,8 @@ private void onFinalResponse(CancellableTask submitTask,
             store.storeFinalResponse(searchTask.getSearchId().getDocId(), threadContext.getResponseHeaders(),response,
                 ActionListener.wrap(resp -> unregisterTaskAndMoveOn(searchTask, nextAction),
                                     exc -> {
-                                        if (exc.getCause() instanceof DocumentMissingException == false) {
+                                        if (exc.getCause() instanceof DocumentMissingException == false &&
+                                                exc.getCause() instanceof VersionConflictEngineException == false) {
```

Comment:
not related to your change, but I think this one should be changed like I did in other places to use `ExceptionHelper.unwrapCause` , I forgot about this one. When can version conflict be thrown again?

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/55435/commits/bf01fc0a6640c82ddcc04af379313e4b950cef38